### DISCUSSION TITLE
How to use clusters with multiple protocols on the same port

### DISCUSSION THREAD

**Discussion by KlemensGenetec** [dotnet/yarp#1459] [Q&A] [answered]
Hello!

I have a question how to handle ports that allow multiple protocols such as http and ws, or https and wss. In the cluster/address I should put in a full URL such as https://xxx/, but then i don't know the protocol that the upstream used. 

So something like
```
"SignalR API": {
  "ClusterId": "my-cluster",
  "Match": {
	  "Path": "/signalr/{**catch-all}"
  }
...
"Clusters": {
	"my-cluster": {
		"Destinations": {
			"1": {
				"Address": "https://XXX"
			}
		}
	}
```

Both http and ws are allowed in the YARP server and should forward to the cluster with the corresponding protocol. What would be the best to achieve that?

**Comment by MihaZupan** [comment, marked as answer]
YARP will proxy both HTTP and WebSocket traffic by default (it will handle the HTTP => WebSocket upgrade for you).

A minimal configuration like you shared is sufficient to enable both protocols.

**Reply by Tratcher** [reply]
The ws and wss schemes tend to only be used on clients. On servers and YARP the http and https schemes are used for both HTTP and WebSockets protocols. Headers are sufficient to distinguish between the two.

**Reply by KlemensGenetec** [reply]
That's great news! Should that be documented somewhere since ws is a pretty common use case?

**Reply by Tratcher** [reply]
Filed a doc issue: https://github.com/microsoft/reverse-proxy/issues/1474
